lib/msal-core/src/cache/AuthCache.ts

```diff
@@ -206,6 +208,24 @@ export class AuthCache extends BrowserStorage {// Singleton
         return results;
     }
 
+    /**
+     * Get idToken from the cache
+     */
+    getIdToken(clientId: string, homeAccountIdentifier: string, authority: string): AccessTokenCacheItem {
+        const idTokenKey: AccessTokenKey = {
+            authority: authority,
+            clientId: clientId,
+            scopes: undefined,
+            homeAccountIdentifier: homeAccountIdentifier
+        };
+
+        const idTokenKeyString = JSON.stringify(idTokenKey);
+        const idToken = this.getItem(idTokenKeyString);
+        const idTokenValue = JSON.parse(idToken) as AccessTokenValue;
+        
+        return new AccessTokenCacheItem(idTokenKey, idTokenValue);
```

Comment:
We should check the expiration and only return if it is not expired similar to the access token scenario

Reply:
Refactored the expiration validation a bit, now we remove the ID Token from the cache and return null if expired over the offset.